src/result.jl

```diff
@@ -361,14 +362,16 @@ function TreeSetColoringResult(
 
     # buffer holds the sum of edge values for subtrees in a tree.
     # For each vertex i, buffer[i] is the sum of edge values in the subtree rooted at i.
-    buffer = Vector{R}(undef, nvertices)
+    # Note that we don't need a buffer is all trees are stars.
+    buffer = all(is_star) ? R[] : Vector{R}(undef, nvertices)
```

Comment:
```suggestion
    buffer = Vector{R}(undef, nvertices)
```
Because the check `isempty(buffer)` is not statically knowable

Reply:
Can we use `Nothing` instead `R[]` for the buffer? It will solve the issue with `isempty` if we replace is with `isnothing`.